hasher-matcher-actioner/lambdas/pdq_matcher.py

```diff
@@ -0,0 +1,89 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+import json
+import logging
+import os
+import pickle
+import boto3
+import datetime
+
+from threatexchange.hashing.pdq_faiss_matcher import PDQFlatHashIndex, PDQMultiHashIndex
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+s3_client = boto3.client("s3")
+sns_client = boto3.client("sns")
+dynamodb = boto3.resource("dynamodb")
```

Comment:
For the future, recommend putting the logging setup into a function and then calling the function under __main__ (or whatever the lambda equivalent is) - you'll never know when you'll want to import the file as a library (usually as part of testing).

Calling getLogger() without context will get you the root logger, and then this code mutate state on that logger - it becomes hard to finely control logging setup when importing this file.

Reply:
This is a good call out. Unfortunately, the lambda bootstrapping process as far as I am aware does not support `main` or any other one-time run setups outside of the global context. We could invoke a function at the start of each request in the handler and have it do setup based on if a global variable is false and then set that variable to true. That is one common pattern I have seen in the past.

For the logger specifically though, we could also look at using the logger AWS provides in AWS Lambda Powertools for Python (https://awslabs.github.io/aws-lambda-powertools-python/core/logger/). That has been support for children and root loggers and can help address that concern too.

Reply:
Sounds good to me!